kroxylicious/src/main/java/io/kroxylicious/proxy/internal/FilterHandler.java

```diff
@@ -39,9 +39,9 @@
     private final String sniHostname;
     private final FilterInvoker invoker;
 
-    public FilterHandler(KrpcFilter filter, long timeoutMs, String sniHostname) {
-        this.filter = Objects.requireNonNull(filter);
-        this.invoker = FilterInvokers.from(filter);
+    public FilterHandler(FilterAndInvoker filter, long timeoutMs, String sniHostname) {
+        this.filter = Objects.requireNonNull(filter.filter());
```

Comment:
The `requireNotNull` should probably be on the record. Though the record should possibly enforce that the filter is not null.

Reply:
Done, added null checking to the record and changed which object is checked in FilterHandler